core/arch/arm/plat-sam/platform_sama7g5.c

```diff
@@ -0,0 +1,960 @@
+	/*
+	 * Matrix DDR configuration is hardcoded here and is difficult to
+	 * generate at runtime. Since this configuration expect the secure
+	 * DRAM to be at start of RAM and 8M of size, enforce it here.
+	 */
+	COMPILE_TIME_ASSERT(CFG_TZDRAM_START == DDR_CS_ADDR);
+	COMPILE_TIME_ASSERT(CFG_TZDRAM_SIZE == 0x800000);
+
+	srtop_setting = MATRIX_SRTOP(0, MATRIX_SRTOP_VALUE_128M);
+	sasplit_setting = MATRIX_SASPLIT(0, MATRIX_SASPLIT_VALUE_8M);
+	ssr_setting = MATRIX_LANSECH_S(0)
+			| MATRIX_RDNSECH_S(0)
+			| MATRIX_WRNSECH_S(0);
+	matrix_configure_slave_security(base, 3, srtop_setting,
+					sasplit_setting, ssr_setting);
+
+	/* 4: APB6: Default */
+
+	/*
+	 * 5: SRAM_P0
+	 * 6: SRAM_P1
+	 */
+	srtop_setting = MATRIX_SRTOP(0, MATRIX_SRTOP_VALUE_128K);
+	sasplit_setting = MATRIX_SASPLIT(0, MATRIX_SASPLIT_VALUE_64K);
+	ssr_setting = MATRIX_LANSECH_NS(0)
+			| MATRIX_RDNSECH_S(0)
+			| MATRIX_WRNSECH_S(0);
+	matrix_configure_slave_security(base, 5, srtop_setting,
+					sasplit_setting, ssr_setting);
+	matrix_configure_slave_security(base, 6, srtop_setting,
+					sasplit_setting, ssr_setting);
+
+	/*
+	 * 7: SMC
+	 * EBI_CS0 ----> Slave Region 0
+	 * EBI_CS1 ----> Slave Region 1
+	 * EBI_CS2 ----> Slave Region 2
+	 * EBI_CS3 ----> Slave Region 3
+	 * NFC_CMD ----> Slave Region 4 : Non-Secure
+	 */
+	srtop_setting =	MATRIX_SRTOP(4, MATRIX_SRTOP_VALUE_128M);
+	sasplit_setting = MATRIX_SASPLIT(4, MATRIX_SASPLIT_VALUE_128M);
+	ssr_setting = MATRIX_LANSECH_NS(4)
+			| MATRIX_RDNSECH_NS(4)
+			| MATRIX_WRNSECH_NS(4);
+	matrix_configure_slave_security(base, 7, srtop_setting,
+					sasplit_setting, ssr_setting);
+
+	/* 8: NFC_RAM
+	 * Slave area below SASSPLIT boundary is configured as Not Secured
+	 */
+	srtop_setting = MATRIX_SRTOP(0, MATRIX_SRTOP_VALUE_8K);
+	sasplit_setting = MATRIX_SASPLIT(0, MATRIX_SASPLIT_VALUE_8K);
+	ssr_setting = MATRIX_LANSECH_NS(0);
+	matrix_configure_slave_security(base, 8, srtop_setting,
+					sasplit_setting, ssr_setting);
+
+	/* 9: USB_RAM
+	 * Slave area below SASSPLIT boundary is configured as Not Secured
+	 */
+	srtop_setting = MATRIX_SRTOP(0, MATRIX_SRTOP_VALUE_1M)
+			| MATRIX_SRTOP(1, MATRIX_SRTOP_VALUE_1M)
+			| MATRIX_SRTOP(2, MATRIX_SRTOP_VALUE_4K)
+			| MATRIX_SRTOP(3, MATRIX_SRTOP_VALUE_4K);
+	sasplit_setting = MATRIX_SASPLIT(0, MATRIX_SASPLIT_VALUE_1M)
+			| MATRIX_SASPLIT(1, MATRIX_SASPLIT_VALUE_1M)
+			| MATRIX_SASPLIT(2, MATRIX_SASPLIT_VALUE_4K)
+			| MATRIX_SASPLIT(3, MATRIX_SASPLIT_VALUE_4K);
+	ssr_setting = MATRIX_LANSECH_NS(0)
+			| MATRIX_LANSECH_NS(1)
+			| MATRIX_LANSECH_NS(2)
+			| MATRIX_LANSECH_NS(3);
+	matrix_configure_slave_security(base, 9, srtop_setting,
+					sasplit_setting, ssr_setting);
+}
+
+static void matrix_init(void)
+{
+	matrix_write_protect_disable(matrix_base(MATRIX_SAMA7G54));
+	matrix_configure_slave();
+}
+
+register_phys_mem_pgdir(MEM_AREA_IO_SEC, TZC_BASE_ADDRESS, CORE_MMU_PGDIR_SIZE);
+
+static void tzc400_init(void)
+{
+	struct tzc_region_config cfg = {};
```

Comment:
s/`{}`/`{ }`/  (with a space char in between braces)

Reply:
Updated.